### PR TITLE
CB-5738 api for tokens

### PR DESCRIPTION
**Pull request by yagudin10**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMController.java [hunk 1]
```diff
@@ -230,4 +230,31 @@ void deleteAllObjectPermissions(
...
+     * @param period    period of validity
+     * @return API token info (including a private key)
+     */
+    @NotNull
+    SMAPIToken createAPIToken(@NotNull String tokenName, int period) throws DBException;
+
+    /**
+     * Removes info about API token for a user.
+     */
+    void deleteAPIToken(@NotNull String tokenName) throws DBException;
```
Comment (alexander-skoblikov) [inline plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMController.java]:
```suggestion
    void deleteAPIToken(@NotNull String tokenId) throws DBException;
```

File: plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMController.java [hunk 2]
```diff
@@ -230,4 +230,31 @@ void deleteAllObjectPermissions(
...
+     * Decodes token and gets user id from token.
+     */
+    @NotNull
+    String getUserIdFromAPIToken(@NotNull String token) throws DBException;
+
+    /**
+     * Returns all info about user API tokens that stored in database.
+     */
+    @NotNull
+    List<SMAPIToken> getAPITokens() throws DBException;
```
Comment (alexander-skoblikov) [inline plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMController.java]:
```suggestion
    List<SMAPIToken> getCurrentUserAPITokens() throws DBException;
```

File: plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMController.java [hunk 3]
```diff
@@ -230,4 +230,31 @@ void deleteAllObjectPermissions(
 
     @NotNull
     String[] getTeamMembers(String teamId) throws DBException;
+
+    /**
+     * Decodes token and gets user id from token.
+     */
+    @NotNull
+    String getUserIdFromAPIToken(@NotNull String token) throws DBException;
```
Comment (alexander-skoblikov) [inline plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMController.java]: do we really need this method in non admin interface?

File: plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMAPIToken.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.security;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+
+public record SMAPIToken(@NotNull String tokenName, @Nullable String token, int periodDays, @NotNull String createTime) {
```
Comment (alexander-skoblikov) [inline plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMAPIToken.java]: lets change periodDays -> expireTime and LocalDateTime

File: plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMController.java (hunk 1 above)
Comment (alexander-skoblikov) [reply plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMController.java]: or we really want to delete token by name?

Comment (yagudin10) [reply plugins/org.jkiss.dbeaver.model.sm/src/org/jkiss/dbeaver/model/security/SMController.java]: we need to delete token by name

**Review by alexander-skoblikov — APPROVED** [review]

**Pull request merged**